src/reader.cpp

```diff
@@ -438,56 +365,47 @@ Entry Reader::getEntryFromEncodedPath(const std::string& path) const
 
 Entry Reader::getEntryFromTitle(const std::string& title) const
 {
-  if (!this->zimFileHandler) {
+  if (!this->zimArchive) {
     throw NoEntry();
   }
 
-  auto article = this->zimFileHandler->getArticleByTitle('A', title);
-  if (!article.good()) {
-    throw NoEntry();
-  }
-
-  return article;
-}
-
-/* Return the mimeType without the content */
-bool Reader::getMimeTypeByUrl(const string& url, string& mimeType) const
-{
   try {
-    auto entry = getEntryFromPath(url);
-    mimeType = entry.getMimetype();
-    return true;
-  } catch (NoEntry& e) {
-    mimeType = "";
-    return false;
+    return zimArchive->getEntryByTitle(title);
+  } catch(zim::EntryNotFound& e) {
+    throw NoEntry();
   }
-
 }
 
 bool Reader::pathExists(const string& path) const
 {
-  if (!zimFileHandler)
+  if (!zimArchive)
   {
     return false;
   }
 
-  char ns = 0;
-  string titleStr;
-  _parseUrl(path, &ns, titleStr);
-  zim::File::const_iterator findItr = zimFileHandler->find(ns, titleStr);
-  return findItr != zimFileHandler->end() && findItr->getUrl() == titleStr;
+  return zimArchive->hasEntryByPath(path);
 }
 
 /* Does the ZIM file has a fulltext index */
 bool Reader::hasFulltextIndex() const
 {
-  if (!zimFileHandler || zimFileHandler->is_multiPart() )
+  if (!zimArchive)
   {
     return false;
   }
 
-  return ( pathExists("Z//fulltextIndex/xapian")
-        || pathExists("X/fulltext/xapian"));
+  for(auto path: {"Z//fulltextIndex/xapian", "X/fulltext/xapian"}) {
```

Comment:
Is the double slash in the path fine?

Reply:
Yes, it is a bug in the old zim files. Recent files uses `X/fulltext/xapian`.